packages/datatrak-web/src/features/Tasks/TaskDetails/TaskDetails.tsx

```diff
@@ -110,23 +105,31 @@ export const TaskDetails = ({ task }: { task: SingleTaskResponse }) => {
     control,
     handleSubmit,
     watch,
-    register,
     formState: { dirtyFields },
     reset,
   } = formContext;
 
-  const navigateBack = () => {
-    navigate(backLink || ROUTES.TASKS);
-  };
-
-  const { mutate: editTask, isLoading: isSaving } = useEditTask(task.id, navigateBack);
+  const { mutate: editTask, isLoading: isSaving } = useEditTask(task.id);
 
   const isDirty = Object.keys(dirtyFields).length > 0;
 
   const onClearEdit = () => {
     reset();
   };
 
+  // Reset form when task changes, i.e after task is saved and the task is re-fetched
+  useEffect(() => {
```

Comment:
I think that it would be cleaner to use the onSucess call back from the useEdit hook. useEffects tend to be hard to work with and I don't think it's needed here

Reply:
I think it still needs to be a `useEffect` because after the task is updated it clears the query cache and reloads it, which will then change the default values. If we do it in the `onSuccess` handler we would need to change the endpoint to return differently to the rest of the endpoints (with the updated task) which I think is arguably a worse option than using a `useEffect`.

Reply:
Instead of changing the edit response, you can just return the task payload from the useEditTask hook and use that in the onSuccess callback. However it is pretty academic so I am happy either way.